bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusOhmpilotHandler.java

```diff
@@ -0,0 +1,136 @@
+import org.openhab.binding.fronius.internal.api.OhmpilotRealtimeResponseDTO;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Thing;
+
+/**
+ * The {@link FroniusOhmpilotHandler} is responsible for updating the data, which are
+ * sent to one of the channels.
+ *
+ * @author Hannes Spenger - Initial contribution
+ *
+ */
+public class FroniusOhmpilotHandler extends FroniusBaseThingHandler {
+
+    private OhmpilotRealtimeBodyDataDTO ohmpilotRealtimeBodyData;
+    private FroniusBaseDeviceConfiguration config;
+
+    public FroniusOhmpilotHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected String getDescription() {
+        return "Fronius Ohmpilot";
+    }
+
+    @Override
+    public void refresh(FroniusBridgeConfiguration bridgeConfiguration) {
+        updateData(bridgeConfiguration, config);
+        updateChannels();
+        updateProperties();
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FroniusBaseDeviceConfiguration.class);
+        super.initialize();
+    }
+
+    /**
+     * Update the channel from the last data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     * @return the last retrieved data
+     */
+    @Override
+    protected Object getValue(String channelId) {
+        if (ohmpilotRealtimeBodyData == null) {
+            return null;
+        }
+
+        final String[] fields = channelId.split("#");
+        if (fields.length < 1) {
+            return null;
+        }
+        final String fieldName = fields[0];
+
+        switch (fieldName) {
+            case FroniusBindingConstants.OHMPILOT_POWER_REAL_SUM:
+                return new QuantityType<>(ohmpilotRealtimeBodyData.getPowerPACSum(), Units.WATT);
+            case FroniusBindingConstants.OHMPILOT_ENERGY_REAL_SUM_CONSUMED:
+                return new QuantityType<>(ohmpilotRealtimeBodyData.getEnergyRealWACSumConsumed(), Units.WATT_HOUR);
+            case FroniusBindingConstants.OHMPILOT_ENERGY_SENSOR_TEMPERATURE_CHANNEL_1:
+                return new QuantityType<>(ohmpilotRealtimeBodyData.getTemperatureChannel1(), Units.KELVIN);
+            case FroniusBindingConstants.OHMPILOT_STATE_CODE:
+                return new QuantityType<>(ohmpilotRealtimeBodyData.getStateCode(), Units.ONE);
+            case FroniusBindingConstants.OHMPILOT_ERROR_CODE:
+                return new QuantityType<>(ohmpilotRealtimeBodyData.getErrorCode(), Units.ONE);
+
+            default:
+                break;
+        }
+
+        return null;
+    }
+
+    private void updateProperties() {
+        if (ohmpilotRealtimeBodyData == null) {
+            return;
+        }
+
+        Map<String, String> properties = editProperties();
+
+        properties.put(FroniusBindingConstants.OHMPILOT_MODEL, ohmpilotRealtimeBodyData.getDetails().getModel());
```

Comment:
Please replace `FroniusBindingConstants.OHMPILOT_MODEL` by `Thing.PROPERTY_MODEL_ID`
Constant OHMPILOT_MODEL can be removed.
For the same reason, please remove constant METER_MODEL (and use Thing.PROPERTY_MODEL_ID).